lumicks/pylake/kymotracker/detail/peakfinding.py

```diff
@@ -97,13 +97,166 @@ def flatten(self):
         return coordinates, time_points, peak_amplitudes
 
 
+def bounds_to_centroid_data(index_array, left_edges, right_edges):
+    """Helper function to return selection indices, pixel centers and weights.
+
+    This function generates indices for sampling based on the left-most and right-most bound. It
+    also generates appropriate pixel centers and weights that account for the fact that they are
+    only partial pixels.
+
+    Parameters
+    ----------
+    index_array: np.ndarray
+        2D pixel indices
```

Comment:
I think this could be a bit clearer. I'm guessing this is for the whole kymograph, not one scan line at a time? Maybe indicated the expected shape like you do in the `Returns` section could help

Reply:
Yeah. Prior to vectorization it was literally one spot for one scan line at a time. After it's every spot at the same time. I changed it to:
```
        2D pixel indices with shape `(N, M)` where `N` is the number of pixels being refined and
        `M` is the window size used for refinement.
```